tamr_toolbox/data_io/cloud.py

```diff
@@ -0,0 +1,220 @@
+import tempfile
```

Comment:
To appropriately type hint with optional dependencies you will need to include the following at the top of this file (after imports): 
```python
# Building our documentation requires access to all dependencies, including optional ones
# This environments variable is set automatically when `invoke docs` is used
BUILDING_DOCS = os.environ.get("TAMR_TOOLBOX_DOCS") == "1"
if BUILDING_DOCS:
    # Import relevant optional dependencies
    from google.cloud.storage import Client as GcsClient
    from mypy_boto3_s3.client import S3Client
```

Reply:
What type's should I add for the generic upload/download functions?

Reply:
I was also wondering if adding the condition is necessary because with the condition in place, the condition causes NameError when testing for dependencies.